### ISSUE TITLE
`BufEnter` is not executed with `:lua vim.cmd('edit /')`.

### ISSUE THREAD

**Issue by tamago324** [neovim/neovim#13726]
- `nvim --version`: 
```
NVIM v0.5.0-dev+1012-gfe1ebea33
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az207-568

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/runner/work/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

Run :checkhealth for more info
```

- `vim -u DEFAULTS` (version: ) behaves differently?

- Operating system/version: 
```sh
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
```

- Terminal name/version: `wsltty 3.3.0`
- `$TERM`: `xterm-256color`

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
# Alternative for shell-related problems:
# env -i TERM=ansi-256color "$(which nvim)"
```

1. `:lua vim.cmd('edit /')`

### Actual behaviour

It seems that `BufEnter` is not executed when a file path with a slash ending is executed by the `:edit` command in lua's `vim.cmd()` function.

But with `:edit /`, `BufEnter` will be executed correctly.

It would be great if `vim.cmd('edit /')` would execute `BufEnter`, although this can be handled by doing the following

```lua
vim.cmd('edit /')
vim.cmd('doautocmd BufEnter')
```

### Expected behaviour

I want `BufEnter` to be executed when `:lua vim.cmd('edit /')` is executed.

**Comment by jamessan** [comment]
Duplicate of #13711

**Comment by zeertzjq** [comment]
This is not a duplicate of #13711. #17846 fixes this one, but not #13711.
